### PR TITLE
Pkg3: rename "Manifest" => "Lockfile"

### PR DESCRIPTION
**Pull request by StefanKarpinski**
I'm not actually sure how I feel about this rename. On the one hand, "lockfile" seems to have become a standard term of art for this kind of file in package managers. On the other hand, the name "lock file" already means something, so that name choice is kind of unfortunate:

* https://fileinfo.com/extension/lock
* https://unix.stackexchange.com/questions/143676/why-do-package-managers-need-lock-files

The term "manifest" is used by Cargo and other package managers to mean something else entirely – namely a description of which versions of other packages a project is compatible with. Of course, that's not at all what the English word "manifest" means – a manifest is a detail list of the contents of something. In other words, exactly what this kind of file is. Sigh. This is why we can't have nice things.

### CODE REVIEW

**Comment by JeffBezanson** [comment]
I think we can and should call it what we want to.

**Comment by swissr** [comment]
(Didn't look into Pkg3 other than browsing the JULEP and youtube). `Lockfile.toml` would lead me to a wrong track, I wouldn't expect to find a listing of library/package/versions/dependencies in it. Hmm, I was not aware of the lockfile term used "for this kind of file in package managers".

With `Manifest.toml`  I would expect some kind of a listing. No idea what, but being abstract I wouldn't expect the rust cargo manifest to be the same as the Julia manifest (or the one for Windows apps, ...).

Not sure if the file that "records the exact versions of each of those dependenices[*] and their transitive dependencies" could be given the more concrete name `Dependencies.toml`? But then `Project.toml` also has (less detailed) dependencies and the name leads to confusion? From a quick glance over the changes, I think 'dependencies' could work (`status [-d|--dependencies] is a bit unexpected). But this said, I'd prefer Manifest.toml.

[*] btw, here's a typo in the README.md

File: stdlib/Pkg3/README.md [hunk 1]
```diff
@@ -36,14 +36,14 @@ INFO: Resolving package versions
...
-INFO: Updating manifest file /Users/stefan/projects/julia/Manifest.toml
+INFO: Updating lockfile file /Users/stefan/projects/julia/Lockfile.toml
 INFO:  [34da2185] ↓ Compat v0.31.0 ⇒ v0.28.0
 
 pkg> rm Compat
 INFO: Updating project file /Users/stefan/projects/julia/Project.toml
 INFO:  [-] Compat = "34da2185-b29b-5c13-b0c7-acf172513d20"
 INFO:  [-] JSON = "682c06a0-de6a-54ab-a142-c8b1cf79cde6"
-INFO: Updating manifest file /Users/stefan/projects/julia/Manifest.toml
+INFO: Updating lockfile file /Users/stefan/projects/julia/Lockfile.toml
```
Comment (nalimilan) [inline stdlib/Pkg3/README.md]: "lockfile file"?

Comment (StefanKarpinski) [reply stdlib/Pkg3/README.md]: It's just a straw man PR to see how we feel about it.

**Comment by ViralBShah** [comment]
I think `Manifest` is pretty self explanatory. `Lockfile` is strictly worse and I wish people handn't done that. I agree with @JeffBezanson that we can call things whatever we want.

**Comment by garrison** [comment]
Python also has `MANIFEST.in` files, which mean something else entirely.  I'd be in favor of a better name than `Manifest.toml`, but I don't think `Lockfile.toml` is it.

**Comment by StefanKarpinski** [comment]
I think `Manifest.toml` is a perfect name for this aside from the fact that other languages have used it in ways that are incompatible with what the word "manifest" means 😬.

**Comment by JeffBezanson** [comment]
Allow me to help out here.

**Pull request closed without merging**